pkg/utils/k8sutil.go

```diff
@@ -565,7 +565,8 @@ func UpdateK8sCustomResource(f *spec.Function) error {
 	}
 
 	// TODO: looking for a way to not writing to temp file
-	err = ioutil.WriteFile(".func.json", funcJSON, 0644)
+	filename := os.TempDir() + "/.func.json"
+	err = ioutil.WriteFile(filename, funcJSON, 0644)
 	if err != nil {
 		return err
```

Comment:
All these early-return paths will leak the tempfile.  Should cleanup via `defer` immediately after the tempdir/file is created.